common_content_filter/test/test_api.cpp

```diff
@@ -0,0 +1,153 @@
+  }
+
+protected:
+  void set_options()
+  {
+    rcutils_allocator_t allocator = rcutils_get_default_allocator();
+    rmw_subscription_content_filter_options_t options =
+      rmw_get_zero_initialized_content_filter_options();
+    EXPECT_EQ(
+      RMW_RET_OK,
+      rmw_subscription_content_filter_options_init(
+        filter_expression,
+        expression_parameter_argc,
+        expression_parameter_argv,
+        &allocator,
+        &options)
+    );
+    OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
+    {
+      rmw_subscription_content_filter_options_fini(&options, &allocator);
+    });
+
+    EXPECT_TRUE(common_content_filter_set(instance, &options));
+  }
+
+  void * instance;
+  const rosidl_message_type_support_t * type_support;
+  const char * filter_expression = "int32_value = %0";
+  size_t expression_parameter_argc = 1;
+  const char * expression_parameter_argv[1] = {"4"};
+};
+
+TEST_F(TestCommonContentFilterAPI, is_enabled) {
+  EXPECT_FALSE(common_content_filter_is_enabled(nullptr));
+  EXPECT_FALSE(common_content_filter_is_enabled(instance));
+  set_options();
+  EXPECT_TRUE(common_content_filter_is_enabled(instance));
+}
+
+TEST_F(TestCommonContentFilterAPI, evaluate) {
+  EXPECT_FALSE(common_content_filter_evaluate(nullptr, nullptr, false));
+  EXPECT_FALSE(common_content_filter_evaluate(instance, nullptr, false));
+  EXPECT_FALSE(common_content_filter_evaluate(nullptr, nullptr, true));
+  EXPECT_FALSE(common_content_filter_evaluate(instance, nullptr, true));
+
+  test_msgs__msg__BasicTypes msg;
+  test_msgs__msg__BasicTypes__init(&msg);
+  msg.int32_value = 3;
+  OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
+  {
+    test_msgs__msg__BasicTypes__fini(&msg);
+  });
+
+  // test serialized message
+  rmw_serialized_message_t serialized_message =
+    rmw_get_zero_initialized_serialized_message();
+  rcutils_allocator_t allocator = rcutils_get_default_allocator();
+  EXPECT_EQ(
+    RMW_RET_OK,
+    rmw_serialized_message_init(&serialized_message, 1U, &allocator)
+  );
+  OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
+  {
+    rmw_serialized_message_fini(&serialized_message);
+  });
+
+  EXPECT_EQ(
+    RMW_RET_OK,
+    rmw_serialize(&msg, type_support, &serialized_message)
+  );
+
+  // no filter setting, expect true
+  EXPECT_TRUE(common_content_filter_evaluate(instance, &msg, false));
+  EXPECT_TRUE(common_content_filter_evaluate(instance, &serialized_message, true));
+  // after setting filter with "int32_value = 4"
+  set_options();
+  // expect msg(int32_value = 3) return false
+  EXPECT_FALSE(common_content_filter_evaluate(instance, &msg, false));
+  EXPECT_FALSE(common_content_filter_evaluate(instance, &serialized_message, true));
+  // update msg with 4
+  msg.int32_value = 4;
+  EXPECT_EQ(
+    RMW_RET_OK,
+    rmw_serialize(&msg, type_support, &serialized_message)
+  );
+
+  EXPECT_TRUE(common_content_filter_evaluate(instance, &msg, false));
+  EXPECT_TRUE(common_content_filter_evaluate(instance, &serialized_message, true));
```

Comment:
Do we need a `false` test case ?

Reply:
The `false` test case located at https://github.com/iuhilnehc-ynos/common_content_filter/blob/e7957a0df6fee7c2baef732b3742da34b5e234f2/common_content_filter/test/test_api.cpp#L114-L115